internal/raft/replicator.go

```diff
@@ -76,10 +77,39 @@ func (this *replicator) Start() {
 	go this.node.purgeFile()
 }
 
-func (this *replicator) ListMembers() (uint64, map[uint64]string) {
-	lead := this.node.getLeaderId()
-	peers := this.node.rpeers
-	return lead, peers
+func (repl *replicator) ListMembers() (uint64, map[uint64]*models.NodeInfo) {
+	lead := repl.node.getLeaderId()
+	members := make(map[uint64]*models.NodeInfo)
+	for id, url := range repl.node.rpeers {
+		activeSince := repl.node.transport.ActiveSince(types.ID(id))
+		nodeInfo := models.NodeInfo{
+			NodeUrl: url,
+			NodeId:  id,
+		}
+		if id == lead {
+			nodeInfo.Status = models.NodeInfo_LEADER
+		} else if activeSince.IsZero() {
+			nodeInfo.Status = models.NodeInfo_OFFLINE
+		} else if id == repl.node.id {
+			//get current node status.
+			status := repl.node.node.Status()
+			if status.RaftState.String() == "StateFollower" {
```

Comment:
Please extract `status.RaftState.String()` to a variable.

Reply:
Done